### PR TITLE
Abort on timeout.

### PR DESCRIPTION
**Pull request by arvenil**
# Abort data collection: 
* when connection gets closed
* on prometheus scrape timeout `X-Prometheus-Scrape-Timeout-Seconds`
  * give exporter time to finish and send data with `--timeout-offset`. Default `0.1` second.

Tested manually. Returns metrics even if scraping was canceled. Example errors:
```console
ERRO[0014] Error scraping for collect.global_status: context deadline exceeded  source="exporter.go:104"
ERRO[0014] Error scraping for collect.info_schema.innodb_metrics: context deadline exceeded  source="exporter.go:104"
```

Based on https://github.com/prometheus/blackbox_exporter/blob/master/main.go#L74-L83

Let me know what you think.

### CODE REVIEW

**Comment by brian-brazil** [comment]
In the blackbox exporter we use a flag-based constant offset, we should
probably be consistent.

**Comment by arvenil** [comment]
Indeed. Thanks for pointing this out, I will change it.

**Comment by arvenil** [comment]
Fixed.

File: mysqld_exporter.go [hunk 1]
```diff
@@ -29,6 +32,10 @@ var (
 		"web.telemetry-path",
 		"Path under which to expose metrics.",
 	).Default("/metrics").String()
+	timeoutOffset = kingpin.Flag(
+		"timeout-offset",
+		"Offset to subtract from timeout in seconds.",
+	).Default("0.5").Float64()
```
Comment (SuperQ) [inline mysqld_exporter.go]: Should we make this `0.1` if people are going to be scraping with 1s intervals for some collectors?

Comment (arvenil) [reply mysqld_exporter.go]: This is why originally I went with percentage of timeout. But I think 0.5 makes more sense. It means, the collector has `0.5s` to stop collecting and send data collected so far to prometheus server. So it's more like a constant that depends on tcp connection and how long it takes for exporter to stop collecting.

But I'm simply ok with any version that works.

Comment (arvenil) [reply mysqld_exporter.go]: What do you think @SuperQ ? Should I change it to `0.1`? It's not a big issue for us for now, we are mostly interested in any form of aborting data collection after timeout - so let me know if you would like any changes here.

**Comment by arvenil** [comment]
Tested manually. Returns metrics even if scraping was canceled. Example errors:
```console
ERRO[0014] Error scraping for collect.global_status: context deadline exceeded  source="exporter.go:104"
ERRO[0014] Error scraping for collect.info_schema.innodb_metrics: context deadline exceeded  source="exporter.go:104"
```

File: mysqld_exporter.go (hunk 1 above)
Comment (SuperQ) [reply mysqld_exporter.go]: How about we split the difference, make it `0.25`. I think 250ms should be enough cleanup and transmit time for most use cases.

Comment (arvenil) [reply mysqld_exporter.go]: Makes perfect sense. Done.

**Comment by arvenil** [comment]
@SuperQ Worth merging or should I close it?

**Comment by SuperQ** [comment]
@arvenil Sorry, busy month, it just fell off my review TODO list. I'll take a look at it on Monday.

**Pull request merged**